gpflow/base.py

```diff
@@ -140,9 +147,30 @@ def __init__(
             value = _cast_to_dtype(value, dtype)
 
         _validate_unconstrained_value(value, transform, dtype)
-        super().__init__(value, transform, dtype=value.dtype, trainable=trainable, name=name)
 
-        self.prior = prior  # type: Optional[Prior]
+        if shape is not None:
+            assert (
+                pretransformed_shape is None
+            ), "Cannot set both `shape` and `pretransformed_shape`."
+            assert transformed_shape is None, "Cannot set both `shape` and `transformed_shape`."
+            pretransformed_shape = shape
+            transformed_shape = shape
+
+        super().__init__(
+            value,
+            transform,
+            dtype=value.dtype,
+            trainable=trainable,
+            name=name,
+            shape=transformed_shape,
```

Comment:
Is this the right way round? Looking at `TransformedVariable.__init__` suggests that `shape` gets passed to the _pretransformed_ variable as part of kwargs, whie `self._shape` is the calculated shape of the _transformed_ value. Or am I misreading it?

Reply:
> Is this the right way round? Looking at `TransformedVariable.__init__` suggests that `shape` gets passed to the _pretransformed_ variable as part of kwargs, whie `self._shape` is the calculated shape of the _transformed_ value. Or am I misreading it?

You're right. Those have been swapped. I'll swap them (back).